tests/registry_disk_test.go

```diff
@@ -85,6 +86,22 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 		Expect(disksFound).To(Equal(1))
 	}
 
+	table.DescribeTable("should", func(image string, policy k8sv1.PullPolicy, expectedPolicy k8sv1.PullPolicy) {
+		vmi := tests.NewRandomVMIWithEphemeralDisk(image)
+		vmi.Spec.Volumes[0].ContainerDisk.ImagePullPolicy = policy
+		vmi = tests.RunVMIAndExpectScheduling(vmi, 60)
+		Expect(vmi.Spec.Volumes[0].ContainerDisk.ImagePullPolicy).To(Equal(expectedPolicy))
+		pod := tests.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+		container := tests.GetContainerDiskContainerOfPod(pod, vmi.Spec.Volumes[0].Name)
+		Expect(container.ImagePullPolicy).To(Equal(expectedPolicy))
```

Comment:
delete vmi here after check is complete or are we depending on the "before cleanup" to handle this?

Reply:
Yes we rely on the cleanup.